zio-http/src/main/scala/zhttp/core/Util.scala

```diff
@@ -21,4 +21,33 @@ object Util {
        |</html>
        |""".stripMargin
   }
+
+  def listFiles(dirPath: java.nio.file.Path): String = {
+    val buf = new StringBuilder()
+      .append("<!DOCTYPE html>\r\n")
+      .append("<html><head><meta charset='utf-8' /><title>")
+      .append("Listing of: ")
+      .append(dirPath)
+      .append("</title></head><body>\r\n")
+      .append("<h3>Listing of: ")
+      .append(dirPath)
+      .append("</h3>\r\n")
+      .append("<ul>")
+      .append("<li><a href=\"../\">..</a></li>\r\n")
+
+    val files = dirPath.toFile.listFiles()
+    files.foreach(f => {
+      if (f.canRead) {
+        val name = f.getName
+        buf
+          .append("<li><a href=\"")
+          .append(name)
+          .append("\">")
+          .append(name)
+          .append("</a></li>\r\n")
+      }
+    })
+    buf.append("</ul></body></html>\r\n")
+    buf.toString()
+  }
```

Comment:
Should use the built-in templating capabilities — check `HtmlTemplating.scala`

Reply:
Will do this in a separate PR - https://github.com/dream11/zio-http/issues/863